### DISCUSSION TITLE
Trying to compile new benchmarks. (stdio.h error and opencl-c.h)

### DISCUSSION THREAD

**Discussion by JunjoFor** [sarchlab/mgpusim#123] [Q&A]
I am trying to add benchmarks to mgpusim because I need benchmarks that generate big mem traces and work well with multiple gpus. I am trying to compile new benchmarks from rodinia but I am running through error I don't know how to solve. I am using rocm 4.0.1 in ubuntu 20.04 and clang-ocl. I am getting `/usr/include/stdio.h:137:14: error: extern variable must reside in constant address space` and other errors related to this two files. 

Has anyone some advice to solve this problem or how to compile things to add a benchmark?

I am using the option -mcpu=gfx803, because as far I know it to obtain the hsaco of the architecture that simulate mgpusim.

**Comment by syifan** [comment]
It looks like a ROCm compiler issue. Can you try ROCm 3.9 instead?

**Reply by JunjoFor** [reply]
I am using ROCm docker. I tried with 3.9 but I get he same error. I think it might be because some issues with libc (?). Because with the docker clean i get this error at first.

`
/usr/include/stdio.h:27:10: fatal error: 'bits/libc-header-start.h' file not found
#include <bits/libc-header-start.h>
`
Installing gcc-multilib solve this issue but then appear the same issues than before.  I tried Ubuntu 18.04 but is the same.

**Reply by syifan** [reply]
Yes. Using Docker is recommended. My intuition is that there is a compatibility issue between the system environment and the benchmark. Are there any other benchmark you can try compiling. Just to make sure there is no problem in the compilation environment?

**Reply by JunjoFor** [reply]
I tried now with parboil, I have the same issue. The compiler couldn't find a libc header, I also tried installing gcc-multilib and then I had the same compiling errors. I guess it's an liking error in the docker but I have no idea how to solve it or who I should ask.

**Reply by syifan** [reply]
What about writing a hello work C/C++ program without HIP?
